src/main/java/edu/illinois/library/cantaloupe/source/HttpSource.java

```diff
@@ -410,17 +411,21 @@ private void fetchRangedGETResponse() throws IOException {
             try {
                 final HttpClient client = getHTTPClient(info);
 
-                ContentResponse response = client.newRequest(info.getURI())
-                        .timeout(getRequestTimeout(), TimeUnit.SECONDS)
-                        .header("Range", "bytes=0-" + (FORMAT_INFERENCE_RANGE_LENGTH - 1))
-                        .method(HttpMethod.GET)
-                        .send();
+                InputStreamResponseListener listener = new InputStreamResponseListener();
+
+                client.newRequest(info.getURI())
+                    .timeout(getRequestTimeout(), TimeUnit.SECONDS)
+                    .header("Range", "bytes=0-" + (FORMAT_INFERENCE_RANGE_LENGTH - 1))
+                    .method(HttpMethod.GET)
+                    .send(listener);
+
+                Response response = listener.get(getRequestTimeout(), TimeUnit.SECONDS);
+
                 rangedGETResponseStatus = response.getStatus();
                 rangedGETResponseHeaders = response.getHeaders();
-                rangedGETResponseEntity = response.getContent();
-            } catch (ExecutionException e) {
-                throw new AccessDeniedException(info.getURI().toString());
-            } catch (InterruptedException | TimeoutException e) {
+                rangedGETResponseEntity = IOUtils.toByteArray(listener.getInputStream());
```

Comment:
This InputStream needs to be closed I think

Reply:
Done using try-with-resources.